src/connector/src/source/base.rs

```diff
@@ -80,13 +80,78 @@ pub trait SplitEnumerator: Sized {
 pub struct SourceInfo {
     pub actor_id: u32,
     pub source_id: TableId,
+    // There should be a 1-1 mapping between `source_id` & `fragment_id`
+    pub fragment_id: u32,
 }
 
 impl SourceInfo {
-    pub fn new(actor_id: u32, source_id: TableId) -> Self {
+    pub fn new(actor_id: u32, source_id: TableId, fragment_id: u32) -> Self {
         SourceInfo {
             actor_id,
             source_id,
+            fragment_id,
+        }
+    }
+}
+
+// const MAX_BLACKLISTED_STREAM_ERROR_MSGS: usize = 50;
+
+// struct ErrorTruncator {
+//     max_size_reached: bool,
+//     // We will only report a maximum of `MAX_UNIQUE_ERROR_MSGS` unique errors before
+//     // we truncate the error messages to reduce their cardinality
+//     past_error_msgs: Arc<RwLock<HashSet<String>>>
+// }
+
+// impl ErrorTrucator {
+//     fn should_truncate_msg(&self, msg: &str) -> bool {
+//         if !self.past_error_msgs.read().contains(msg) {
+
+//         }
+//         false
+//     }
+// }
+
+#[derive(Debug, Clone)]
+pub struct ErrorReportingContext {
```

Comment:
Perhaps can be `Context` or `SourceContext` to make it shorter. Also implies it's okay to add more things here in the future.

Reply:
prefer `SourceErrorConext`